enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -396,7 +396,7 @@ struct CacheAnalysis {
       return {};
     }
 
-    if (isCertainPrintMallocOrFree(Fn) || isAllocationFunction(*Fn, TLI)) {
+    if (isCertainCacheable(Fn) || isAllocationFunction(*Fn, TLI)) {
```

Comment:
Again this code says that a print/malloc/free function doesn't overwrite a potential load/call/etc and isn't a question about whether the function is cacheable.